services/QuillLMS/app/workers/google_integration/import_classroom_students_worker.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module GoogleIntegration
+  class ImportClassroomStudentsWorker
+    include Sidekiq::Worker
+    sidekiq_options queue: SidekiqQueue::CRITICAL_EXTERNAL
+
+    PUSHER_EVENT = 'google-classroom-students-imported'
+    PUSHER_FAILED_EVENT = 'google-account-reauthorization-required'
+
+    attr_reader :teacher_id
```

Comment:
What's the point of using the `attr_reader` and private method here when the teacher_id just gets passed into `perform`? Couldn't we just do `teacher ||= ::User.find(teacher_id)` in line 14?

Reply:
I usually default to finding of objects in their own memoized method.  That way if different methods call on that object, its found in one place.  However, in this case, you're right to point out that the memoization isn't really buying us much since there's only the `perform` method.